include/optimizer.h

```diff
@@ -24,8 +24,9 @@ class Optimizer {
   /*!
   * \brief Operator constructor, the optimizer is not initialized until the first Update
   * \param opt_type type of the optimizer
+  * \param learning_rate
   */
-  explicit Optimizer(const std::string &opt_type);
+  Optimizer(const std::string &opt_type, mx_float learning_rate);
```

Comment:
I think setters are good.
Sometimes we just want to modify the learning rate to some special layers by hand.